File: Classes/AppWindow.m
// AppWindow.m
//
// Kraftstoff

#import "AppWindow.h"
#import "AppDelegate.h"

@implementation AppWindow

- (void)motionEnded:(UIEventSubtype)motion withEvent:(UIEvent *)event
{
    if (motion == UIEventSubtypeMotionShake) {

        [[NSNotificationCenter defaultCenter]
            postNotification:[NSNotification notificationWithName:kraftstoffDeviceShakeNotification
                                                           object:nil]];
    }
    else
        [super motionEnded:motion withEvent:event];
}

@end

File: Classes/AppWindow.swift
//
//  AppWindow.swift
//  kraftstoff
//
//  Created by Ingmar Stein on 30.04.15.
//
//

import UIKit

class AppWindow: UIWindow {

	override func motionEnded(motion: UIEventSubtype, withEvent event: UIEvent?)	{
		if motion == .MotionShake {
			NSNotificationCenter.defaultCenter().postNotificationName(kraftstoffDeviceShakeNotification, object:nil)
		} else {
			super.motionEnded(motion, withEvent:event)
		}
	}

}
